django/db/migrations/state.py

```diff
@@ -346,6 +346,36 @@ def _reload(self, related_models):
         # Render all models
         self.apps.render_multiple(states_to_be_rendered)
 
+    def resolve_model_field_relations(
+        self, model_key, field_name, field, concretes=None,
+    ):
+        remote_field = field.remote_field
+        if not remote_field:
+            return
+        if concretes is None:
+            concretes, _ = self._get_concrete_models_mapping_and_proxy_models()
+
+        remote_model_key = resolve_relation(remote_field.model, *model_key)
+        if remote_model_key[0] not in self.real_apps and remote_model_key in concretes:
+            remote_model_key = concretes[remote_model_key]
+        self.relations[remote_model_key][model_key].append((field_name, field))
+
+        through = getattr(remote_field, 'through', None)
+        if not through:
+            return
+        through_model_key = resolve_relation(through, *model_key)
+        if through_model_key[0] not in self.real_apps and through_model_key in concretes:
+            through_model_key = concretes[through_model_key]
+        self.relations[through_model_key][model_key].append((field_name, field))
```

Comment:
The 4 lines above look identical to the `remote_model_key` lines a few lines before, except with `through` instead of `remote_field.model`. Maybe that can be a helper method accepting that argument.

Reply:
Good idea, I added an extra hook.